src/components/Forms/type.ts

```diff
@@ -0,0 +1,13 @@
+import { Control, RegisterOptions } from "react-hook-form";
+
+export interface BaseControlledInputProps {
+  id: string;
+  name: string;
+  control: Control<any>;
+  rules?: Exclude<
+    RegisterOptions,
+    "valueAsNumber" | "valueAsDate" | "setValueAs"
```

Comment:
"valueAsNumber" | "valueAsDate" | "setValueAs" does this actually exist in controlled rules?

Reply:
yes it is, here is the reference : https://react-hook-form.com/api/useform/register#options.

But i made a mistake here, the type of rules supposed to not exclude those three props ("valueAsNumber" | "valueAsDate" | "setValueAs") because this is general interface for all controlled input, not only for switch component. Based on reference above, i think the type of rules should be RegisterOptions. What do you think ?

Reply:
waittt kyknya udah bener dehh wkwkwkwkwk

referencenya bukan dari useForm hib. tapi dari Controller.
<img width="928" alt="image" src="https://user-images.githubusercontent.com/93510602/177695072-d3d4f20e-258c-44be-b138-dede60cc8636.png">

rule register semuanya ga ada di situ jadi udah bener harusnya di exclude kayak gitu. gue mistaken ngeliat Exclude dan gue kira itu fungsi buat union gt wkwkkw my bad. bisa di revert ga ya?